packages/ui-components/src/components/windowedlist.ts

```diff
@@ -1524,16 +1692,56 @@ export namespace WindowedList {
    * Windowed list view constructor options
    */
   export interface IOptions<
-    T extends WindowedList.IModel = WindowedList.IModel
+    T = any,
+    U extends WindowedList.IModel<T> = WindowedList.IModel<T>
   > {
     /**
      * Windowed list model to display
      */
-    model: T;
+    model: U;
     /**
      * Windowed list layout
      */
     layout?: WindowedLayout;
+
+    /**
+     * A renderer for the elements of the windowed list.
+     */
+    renderer?: IRenderer<T>;
+
+    /**
+     * Whether the windowed list should display a scrollbar UI.
+     */
+    scrollbar?: boolean;
+  }
+
+  /**
+   * A windowed list element renderer.
+   */
+  export interface IRenderer<T = any> {
+    /**
+     * Create the outer, root element of the windowed list.
+     */
+    createOuter(): HTMLElement;
+
+    /**
+     * Create the virtual scrollbar element.
+     */
+    createScrollbar(): HTMLOListElement;
+
+    /**
+     * Create an individual item rendered in the scrollbar.
+     */
+    createScrollbarItem(
+      index: number,
+      item: T,
```

Comment:
If the `itemsList` is not provided, `item` cannot be defined.

```suggestion
      item?: T,
```

Reply:
`createScrollbarItem` is only called if the list exists and is populated.
```ts
const list = viewModel.itemsList;
if (list?.length) {
  for (let index = 0; index < list.length; index += 1) {
    const item = renderer.createScrollbarItem(index, list.get(index), this);
    item.dataset.index = `${index}`;
    content.appendChild(item);
  }
}
```

Reply:
Maybe `item: T | undefined`